src/applications/disability-benefits/526EZ/actions/index.js

```diff
@@ -1,64 +1,138 @@
-/* eslint-disable no-unused-vars */
 import { apiRequest } from '../../../../platform/utilities/api';
-import { fetchInProgressForm } from '../../../common/schemaform/save-in-progress/actions';
+import Raven from 'raven-js';
 
-export const ITFStatuses = Object.freeze({
-  active: 'active',
-  claim_received: 'claim_received', // eslint-disable-line camelcase
-  duplicate: 'duplicate',
-  expired: 'expired',
+export const SET_PRESTART_STATUS = 'SET_PRESTART_STATUS';
+export const RESET_PRESTART_STATUS = 'RESET_PRESTART_STATUS';
+export const RESET_PRESTART_DISPLAY = 'RESET_PRESTART_DISPLAY';
+
+export const PRESTART_STATUSES = {
+  notAttempted: 'not-attempted',
   pending: 'pending',
-  incomplete: 'incomplete'
-});
+  none: 'none',
+  expired: 'expired',
+  created: 'created',
+  retrieved: 'retrieved',
+  renewed: 'renewed',
+  notRetrievedSaved: 'not-retrieved-saved',
+  notRetrievedNew: 'not-retrieved-new',
+  notCreated: 'not-created',
+  notRenewed: 'not-renewed',
+};
+
+export const prestartSuccessStatuses = new Set([PRESTART_STATUSES.created, PRESTART_STATUSES.retrieved, PRESTART_STATUSES.renewed]);
+
+export const prestartFailureStatuses = new Set([PRESTART_STATUSES.notCreated, PRESTART_STATUSES.notRenewed, PRESTART_STATUSES.notRetrievedNew, PRESTART_STATUSES.notRetrievedSaved]);
 
+export const prestartPendingStatuses = new Set([PRESTART_STATUSES.none, PRESTART_STATUSES.expired, PRESTART_STATUSES.pending]);
 
-// TODO: replace mock once ITF endpoint setup
-export function submitIntentToFile(formConfig, onChange) {
-  const { intentToFileUrl, formId, migrations, prefillTransformer } = formConfig;
-  let ITFStatus = ITFStatuses.pending;
-  return dispatch => {
+export function setPrestartStatus(status, data) {
+  return {
+    type: SET_PRESTART_STATUS,
+    status,
+    data
+  };
+}
 
-    onChange(ITFStatus);
+export function resetPrestartStatus() {
+  return {
+    type: RESET_PRESTART_STATUS
+  };
+}
 
-    const delay = (ms) => new Promise((resolve, reject) => {
-      ITFStatus = ITFStatuses.active;
-      onChange({ ITFStatus });
+export function resetPrestartDisplay() {
+  return {
+    type: RESET_PRESTART_DISPLAY
+  };
+}
 
-      // TODO: if the backend handles resubmission, this check can be removed
-      if (ITFStatus === ITFStatuses.active) {
-        setTimeout(resolve, ms);
+export function checkITFRequest(dispatch, hasSavedForm) {
+  return apiRequest(
+    '/intent_to_file',
+    null,
+    ({ data }) => {
+      let status;
+      let expirationDate;
+      const itfList = data.attributes.intentToFile;
+      if (!itfList || (Array.isArray(itfList) && itfList.length === 0)) {
```

Comment:
@SMLuthi helped to determine what possible shapes the response data might be, any additional insights appreciated as well.